src/cache/redis.service.ts

```diff
@@ -0,0 +1,48 @@
+import { Logger } from '@nestjs/common';
+import { SetOptions } from '@redis/client';
+import { injectable } from 'inversify';
+import { createClient } from 'redis';
+
+@injectable()
+export class RedisService {
+  private readonly logger = new Logger('RedisService');
+  private readonly redis = createClient({
+    url: 'redis://default:redispassword@localhost:6399/0',
+  });
+
+  constructor() {
+    this.connect();
+  }
```

Comment:
```suggestion
  async onModuleInit() {
    await this.connect();
  }
```
https://docs.nestjs.com/fundamentals/lifecycle-events